### ISSUE TITLE
Terraform graph doesn't change with --module-depth

### ISSUE THREAD

**Issue by t-pascal** [hashicorp/terraform#5596]
I recently noticed that `terraform graph` doesn't look different for modules and doesn't scope/descope modules, even when specifying module-depth as shown in the "Plans and Graphs" section here: https://www.terraform.io/docs/modules/usage.html

For example:

``` bash
terraform graph config/ | wc -l
150

terraform graph --module-depth=1 config/ | wc -l
150

terraform graph --module-depth=0 config/ | wc -l
150
```

I can't really include an example but it should be simple enough to reproduce. If you need more details, let me know and I'll manufacture one.

**Comment by SydOps** [comment]
`-module-depth`, there is only one minus.

**Comment by t-pascal** [comment]
Unfortunately, that doesn't make a whole lot of difference. I've found that terraform tends to honour both single and double minuses (which is good):

~~~
terraform --version
Terraform v0.6.12

terraform graph -module-depth=0 config/ | wc -l
159

terraform graph --module-depth=0 config/ | wc -l
159

terraform graph -module-depth=1 config/ | wc -l
159

terraform graph config/ | wc -l
159
~~~

**Comment by ryansch** [comment]
Just noticed this as well.  The resulting graph doesn't have any module information in it.  It does contain all of the resources from the included modules but doesn't have any module boxes around them.

**Comment by t-pascal** [comment]
Still broken as of 

`
terraform --version
Terraform v0.6.16

terraform graph -module-depth=0 config/ | wc -l
426

terraform graph --module-depth=0 config/ | wc -l
426

terraform graph config/ | wc -l
426

terraform graph -module-depth=1 config/ | wc -l
426

terraform graph -module-depth=2 config/ | wc -l
426

**Comment by t-pascal** [comment]
This issue is marked as a question. I think it's confirmed as a bug.

**Comment by t-pascal** [comment]
As another point of reference, `-module-depth=n` works correctly with `terraform show` in Terraform v0.6.16.

**Comment by hposca** [comment]
Still not working with Terraform v0.7.0 or v0.7.1 .
Everything else works fine, but without module information the generated graph becomes messy and hard to understand :/

**Comment by xsellier** [comment]
Any news on this ?

**Comment by mitchellh** [comment]
The `-module-depth` flag on _graph_ is actually no longer available as of 0.8. The graphs we generate for 0.8 no longer collapse modules into a single graph, meaning that flattening them doesn't make sense any more.

The module depth flag doesn't make a lot of sense either because edges often go in and out of a module and if you collapse it to a single box it'll LOOK like a cycle (when it really isn't since the arrows go to something _in_ the module, not to the module itself).

On the flip side, the graphs generated for Terraform 0.8 are generally much smaller/simpler (except for the NEW infra case) so regardless of this change they should be relatively readable. 

The module depth flag still exists and makes sense for apply and so on.
